rust/src/tests/open.rs

```diff
@@ -118,3 +118,42 @@ fn interact_2gb(ctx: &mut TestContext) {
     nix::sys::uio::pread(fd, &mut buf, offset).unwrap();
     assert_eq!(buf, DATA.as_bytes());
 }
+
+// POSIX states that open should return ELOOP, but FreeBSD returns EMLINK instead
+#[cfg(target_os = "freebsd")]
+crate::test_case! {
+    /// open returns EMLINK when O_NOFOLLOW was specified and the target is a symbolic link
+    open_nofollow
+}
+#[cfg(target_os = "freebsd")]
+fn open_nofollow(ctx: &mut TestContext) {
```

Comment:
This looks good.  It would be a bit more general though if you enabled it for every OS, and change the assertion to `e == EMLINK || e == ELOOP`.